students/ZidanLuo/session08/test_Circle_Class.py

```diff
@@ -0,0 +1,131 @@
+
+from Circle_Class import Circle, Sphere
+
+from math import pi
+
+def test_init():
+    c = Circle(4)
+
+
+def test_radius():
+    c = Circle(4)
+
+    assert c.radius == 4
+    
+def test_diameter():
+    c = Circle(4)
+
+    assert c.diameter == 8
+
+
+def test_set_diameter():
+    c = Circle(7)
+    c.diameter = 16
+
+    assert c.radius == 8
+    assert c.diameter == 16
+
+
+def test_area():
+    c = Circle(5)
+
+    assert c.area == pi * 25
+
+
+def test_str():
+    c = Circle(10)
+
+    assert str(c) == "Circle with radius: 10.000000"
+
+
+def test_repr():
+    c = Circle(10)
+    d = eval(repr(c))
+
+    assert repr(c) == "Circle(10)"
+    assert d == c
+
+
+def test_from_diameter():
+    c = Circle.from_diameter(8)
+
+    assert c.diameter == 8
+    assert c.radius == 4
+
+
+
+def test_add_two_circles():
+    c1 = Circle(2)
+    c2 = Circle(4)
+
+    assert (c1 + c2) == Circle(6)
+
+
+def test_mul():
+    c2 = Circle(4)
+
+    assert (c2 * 3) == Circle(12)
+    assert (3 * c2) == Circle(12)
+
+
+def test_compare():
+    c1 = Circle(2)
+    c2 = Circle(4)
+    c3 = Circle(4)
+
+    assert (c1 > c2) == False
```

Comment:
you eantt o use "is" with True, False, and None:

```
assert (c1 > c2) is False
```
etc.

But other wise, really nice tests